app/models/user.rb

```diff
@@ -7,4 +7,8 @@ class User < ApplicationRecord
   validates :first_name, presence: true
   enum role: { default: 0, admin: 1 }
   has_secure_password
+
+  def bookmarked_videos
+    videos.order(:tutorial_id, :position)
```

Comment:
I believe this follows the user story 
`Only makes one query to the database (no n+1 queries)`, 
not entirely sure if it follows the second check box
`Avoids loading all videos into memory`